editor/editor_command_palette.cpp

```diff
@@ -0,0 +1,287 @@
+	return score * 0.69f;
+}
+
+void EditorCommandPalette::_update_command_search(const String &search_text) {
+	commands.get_key_list(&command_keys);
+	ERR_FAIL_COND(command_keys.is_empty());
+
+	// const String search_text = command_search_box->get_text();
+	const bool empty_search = search_text.is_empty();
+	Map<String, TreeItem *> sections;
+	TreeItem *first_section = nullptr;
+
+	// Filter possible candidates.
+	Vector<CommandEntry> entries;
+	for (int i = 0; i < command_keys.size(); i++) {
+		CommandEntry r;
+		r.key_name = command_keys[i];
+		r.display_name = commands[r.key_name].name;
+		r.shortcut_text = commands[r.key_name].shortcut;
+		if (!empty_search && search_text.is_subsequence_ofi(r.display_name)) {
+			r.score = _score_path(search_text, r.display_name.to_lower());
+			entries.push_back(r);
+		}
+	}
+
+	command_keys.clear();
+
+	TreeItem *root = search_options->get_root();
+	root->clear_children();
+
+	if (entries.size() > 0) {
+		if (!empty_search) {
+			SortArray<CommandEntry, CommandEntryComparator> sorter;
+			sorter.sort(entries.ptrw(), entries.size());
+		}
+
+		const int entry_limit = MIN(entries.size(), 300);
+		for (int i = 0; i < entry_limit; i++) {
+			String section_name = entries[i].key_name.get_slice("/", 0);
+			TreeItem *section;
+
+			if (sections.has(section_name)) {
+				section = sections[section_name];
+			} else {
+				section = search_options->create_item(root);
+
+				if (!first_section) {
+					first_section = section;
+				}
+
+				String item_name = section_name.capitalize();
+				section->set_text(0, item_name);
+
+				sections[section_name] = section;
+				section->set_custom_bg_color(0, search_options->get_theme_color("prop_subsection", "Editor"));
+				section->set_custom_bg_color(1, search_options->get_theme_color("prop_subsection", "Editor"));
+			}
+
+			TreeItem *ti = search_options->create_item(section);
+			String shortcut_text = entries[i].shortcut_text == "None" ? "" : entries[i].shortcut_text;
+			ti->set_text(0, entries[i].display_name);
+			ti->set_metadata(0, entries[i].key_name);
+			ti->set_text_align(1, TreeItem::TextAlign::ALIGN_RIGHT);
+			ti->set_text(1, shortcut_text);
+			Color c = Color(1, 1, 1, 0.5);
+			ti->set_custom_color(1, c);
+		}
+
+		TreeItem *to_select = first_section->get_first_child();
+		to_select->select(0);
+		to_select->set_as_cursor(0);
+		search_options->scroll_to_item(to_select);
+
+		get_ok_button()->set_disabled(false);
+	} else {
+		TreeItem *ti = search_options->create_item(root);
+		ti->set_text(0, TTR("No Matching Command"));
+		ti->set_metadata(0, "");
+		Color c = Color(0.5, 0.5, 0.5, 0.5);
+		ti->set_custom_color(0, c);
+		search_options->deselect_all();
+
+		get_ok_button()->set_disabled(true);
+	}
+}
+
+void EditorCommandPalette::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("add_command", "command_name", "key_name", "callable", "arguments", "shortcut_text"), &EditorCommandPalette::add_command, DEFVAL(varray()), DEFVAL("None"));
```

Comment:
Once again, but `add_command` doesn't need the `arguments` parameter. `Callable.bind()` is available in GDScript too.